service/worker/batcher/activities.go

```diff
@@ -308,3 +332,85 @@ func isDone(ctx context.Context) bool {
 		return false
 	}
 }
+
+func getResetEventIDByType(ctx context.Context, resetType enumspb.ResetType, namespace string, workflowExecution *commonpb.WorkflowExecution, frontendClient workflowservice.WorkflowServiceClient) (int64, error) {
+	switch resetType {
+	case enumspb.RESET_TYPE_FIRST_WORKFLOW_TASK:
+		return getFirstWorkflowTaskEventID(ctx, resetType, namespace, workflowExecution, frontendClient)
+	case enumspb.RESET_TYPE_LAST_WORKFLOW_TASK:
+		return getLastWorkflowTaskEventID(ctx, resetType, namespace, workflowExecution, frontendClient)
+	default:
+		errorMsg := fmt.Sprintf("provided reset type (%v) is not supported.", resetType)
+		return 0, serviceerror.NewInvalidArgument(errorMsg)
+	}
+}
+
+func getLastWorkflowTaskEventID(ctx context.Context, resetType enumspb.ResetType, namespace string, workflowExecution *commonpb.WorkflowExecution, frontendClient workflowservice.WorkflowServiceClient) (workflowTaskEventID int64, err error) {
+	req := &workflowservice.GetWorkflowExecutionHistoryRequest{
+		Namespace:       namespace,
+		Execution:       workflowExecution,
+		MaximumPageSize: 1000,
+		NextPageToken:   nil,
+	}
+	for {
+		resp, err := frontendClient.GetWorkflowExecutionHistory(ctx, req)
+		if err != nil {
+			//TODO: What type of error should we return here?
+			return 0, errors.New("GetWorkflowExecutionHistory failed")
+		}
+		for _, e := range resp.GetHistory().GetEvents() {
+			if e.GetEventType() == enumspb.EVENT_TYPE_WORKFLOW_TASK_COMPLETED {
+				workflowTaskEventID = e.GetEventId()
+			} else if e.GetEventType() == enumspb.EVENT_TYPE_WORKFLOW_TASK_SCHEDULED {
+				workflowTaskEventID = e.GetEventId() + 1
+			}
+		}
+		if len(resp.NextPageToken) != 0 {
+			req.NextPageToken = resp.NextPageToken
+		} else {
+			break
+		}
+	}
+	if workflowTaskEventID == 0 {
+		//TODO: what type of error should we return here?
+		return 0, errors.New("GetLastWorkflowTaskID failed:unable to find any scheduled or completed task")
```

Comment:
Maybe bad request?

Reply:
i think we can set to first workflow task scheduled event

Reply:
This means there is no workflow task scheduled? e.g. cron/retry

Reply:
yes, if we reach this line this means there is no workflow task scheduled event, otherwise we would have set it in the loop.

Reply:
Correct, this means no workflow is scheduled.